### ISSUE TITLE
bazel test syntaxnet error 

### ISSUE THREAD

**Issue by vad23klev** [tensorflow/models#284]
**on** `bazel test --linkopt=-lrt syntaxnet/... util/utf8/...`
have output
`//syntaxnet:beam_reader_ops_test                                         FAILED in 0.4s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/beam_reader_ops_test/test.log
//syntaxnet:graph_builder_test                                           FAILED in 0.4s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/graph_builder_test/test.log
//syntaxnet:lexicon_builder_test                                         FAILED in 0.5s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/lexicon_builder_test/test.log
//syntaxnet:parser_trainer_test                                          FAILED in 0.5s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/parser_trainer_test/test.log
//syntaxnet:reader_ops_test                                              FAILED in 0.5s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log
//syntaxnet:text_formats_test                                            FAILED in 0.4s
  /home/vadim/.cache/bazel/_bazel_vadim/7025f6cc275cf60feb7b64b43813fcae/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/text_formats_test/test.log

Executed 6 out of 12 tests: 6 tests pass and 6 fail locally.
`
in log I see two strange lines:
`
 File "/usr/lib/python3.5/importlib/**init**.py", line 126, in import_module
...
ImportError: No module named '_pywrap_tensorflow'
`
use ArchLinux, bazel 0.2.2b, tensorflow configure with /usr/bin/python2.

**Comment by BayBenj** [comment]
I have this same exact issue.

**Comment by BayBenj** [comment]
I have this exact same issue. This is what I see: 
<img width="1042" alt="tests" src="https://cloud.githubusercontent.com/assets/17600555/17115444/8ddfbdac-5270-11e6-9889-d0527f0dbebb.png">

With this traceback:
<img width="1263" alt="traceback" src="https://cloud.githubusercontent.com/assets/17600555/17115462/9c2d8d80-5270-11e6-890a-2b3f9cce3480.png">


Any suggestions on what we're doing wrong? These are my specs:
I am using Mac OS X.
My Python version is 2.7.11.
I am using bazel version 0.2.2b.
Numpy, asciitree, protobuf 3.0.0b2, and swig are all installed.

**Comment by tfmeads** [comment]
I have this exact problem now too. Running Linux Mint (based on Ubuntu, so it should work)
All dependencies installed, bazel 0.2.2b, python 2.7.11

**Comment by sammoes** [comment]
bazel test syntaxnet/... util/utf8/...
shows a very large number of Info / Warnings
gcc version 5.4.0, cudnn 5, cuda 8
fresh bazel install

**Comment by borisstock** [comment]
Same problem here

**Comment by borisstock** [comment]
Ok, I see now that I did not have the same error. In my case the python lib "mock" was missing. This can be resolved with "pip install mock". mock is a new dependency. Maybe that should be added to the README?

**Comment by sammoes** [comment]
fixed it by manually changing tensorflow/tensorflow/tools/proto_text/BUILD like this:
fayeshine/tensorflow@76f6e0a

**Comment by girving** [comment]
@calberti Can you comment?

**Comment by shivanraptor** [comment]
Exact same issue with Mac OS X Sierra 10.12.3

    Bazel version: 0.4.4-homebrew

Same error as @borisstock as well. After installed `mock` and `asciitree` via `pip install mock` and `pip install asciitree`, the `demo.sh` still has `ImportError: no module named mock` (same for asciitree as well).

    TensorFlow version: 1.0.1

**Comment by AyushiAggarwal** [comment]
Failing tests when running ```bazel test --linkopt=-lrt syntaxnet/... util/utf8/...```
Executed 25 out of 25 tests: 19 tests pass and 6 fail locally 

Screenshot of tests that failed is given below: 
![image](https://user-images.githubusercontent.com/6222373/30131797-ef6ffdd8-9302-11e7-91f6-ca8034190a55.png)

test.log file shows the following error: 
![image](https://user-images.githubusercontent.com/6222373/30131861-2fbab216-9303-11e7-8cf2-47900380dfce.png)

"F external/org_tensorflow/tensorflow/core/framework/variant_op_registry.cc:79] Check failed: existing == nullptr (0x10a0f30 vs. nullptr)Unary VariantDecodeFn for type_name: tensorflow::Tensor already registered
Aborted"

Details of my configuration on CentOS: 
- gcc 6.3.1
- glib 2.12
- Bazel 0.5.4
- python27

Also note that I already have Tensorflow 1.2.1 installed.
How do I resolve this issue?

**Comment by itsmeolivia** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by lifeiteng** [comment]
@AyushiAggarwal Do you have fixed `Check failed: existing == nullptr (0x18b9ea8 vs. nullptr)Unary VariantDecodeFn for type_name: tensorflow::Tensor already registered`?

I build on TF1.6 with GPU, meet the same error
```
bazel test -c opt --config=cuda --define using_cuda_nvcc=true --define using_gcudacc=true --incompatible_load_argument_is_label=false linnet/...
``` 
![2018-04-26 19 16 50](https://user-images.githubusercontent.com/3107839/39302780-69423358-4986-11e8-8488-d4286fa91bf4.png)

**Comment by AyushiAggarwal** [comment]
@lifeiteng My error was fixed when I pointed tensorflow copy to [this commit](https://github.com/tensorflow/tensorflow/commit/2f6f329aac6e4f23ef20f6e37165a235e232e772)
and made changes suggested by @charlesjohannisen in #2355. 

Does the documentation in fix #2355 help?

**Comment by lifeiteng** [comment]
@AyushiAggarwal Just Fix this, by changing `syntaxnet/BUILD` (base on TensorFlow 1.6)
```
         "@com_googlesource_code_re2//:re2",
         "@protobuf_archive//:protobuf",
         "@org_tensorflow//third_party/eigen3",
-    ] + select({
-        "//conditions:default": [
-            "@org_tensorflow//tensorflow/core:framework",
-            "@org_tensorflow//tensorflow/core:lib",
-        ],
-        "@org_tensorflow//tensorflow:darwin": [
-            "@org_tensorflow//tensorflow/core:framework_headers_lib",
-        ],
-    }),
+        "@org_tensorflow//tensorflow/core:framework_headers_lib",
+    ],
```

also [a TensorFlow fix ](https://github.com/tensorflow/tensorflow/commit/36ec749ec79c2313924666a1c5324620e493d0c4)
